application/lib/registration.go

```diff
@@ -84,11 +85,15 @@ type RegistrationManager struct {
 	registeredDecoys *RegisteredDecoys
 	Logger           *log.Logger
 	PhantomSelector  *PhantomIPSelector
+	LivenessTester	 *lt.CachedLivenessTester
+	DefaultTester	 *lt.UncachedLivenessTester
+	IsCached		 bool
```

Comment:
`DefaultTester` and `IsCached` aren't needed and Liveness tester should be changed to an interface type

```go
	Logger           *log.Logger
	PhantomSelector  *PhantomIPSelector
	PhantomSelector  *PhantomIPSelector
	LivenessTester   lt.LivenessTester
}
```

The interface pattern does the same thing behind the scenes and makes everything more explicit.